pkg/controllers/job/plugins/distributed-framework/pytorch/pytorch.go

```diff
@@ -0,0 +1,192 @@
+package pytorch
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	batch "volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	"volcano.sh/volcano/pkg/controllers/job/helpers"
+	pluginsinterface "volcano.sh/volcano/pkg/controllers/job/plugins/interface"
+)
+
+const (
+	// PytorchPluginName is the name of the plugin
+	PytorchPluginName = "pytorch"
+	// DefaultPort is the default port for pytorch
+	DefaultPort = 23456
+	// DefaultMaster is the default task name of master host
+	DefaultMaster = "master"
+	// DefaultWorker is the default task name of worker host
+	DefaultWorker = "worker"
+
+	// EnvMasterPort is the env name of master port
+	EnvMasterPort = "MASTER_PORT"
+	// EnvMasterAddr is the env name of master addr
+	EnvMasterAddr = "MASTER_ADDR"
+	// EnvWorldSize is the env name of world size
+	EnvWorldSize = "WORLD_SIZE"
+	// EnvWorldSize is the env name of rank
+	EnvRank = "RANK"
+)
+
+type pytorchPlugin struct {
+	pytorchArguments []string
+	clientset        pluginsinterface.PluginClientset
+	masterName       string
+	workerName       string
+	port             int
+}
+
+// New creates pytorch plugin.
+func New(client pluginsinterface.PluginClientset, arguments []string) pluginsinterface.PluginInterface {
+	pp := pytorchPlugin{pytorchArguments: arguments, clientset: client}
+	pp.addFlags()
+	return &pp
+}
+
+func (pp *pytorchPlugin) addFlags() {
+	flagSet := flag.NewFlagSet(pp.Name(), flag.ContinueOnError)
+	flagSet.StringVar(&pp.masterName, "master", DefaultMaster, "name of master role task")
+	flagSet.StringVar(&pp.workerName, "worker", DefaultWorker, "name of worker role task")
+	flagSet.IntVar(&pp.port, "port", DefaultPort, "open port for containers")
+	if err := flagSet.Parse(pp.pytorchArguments); err != nil {
+		klog.Errorf("plugin %s flagset parse failed, err: %v", pp.Name(), err)
+	}
+}
+
+func (pp *pytorchPlugin) Name() string {
+	return PytorchPluginName
+}
+
+func (pp *pytorchPlugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
+	taskType := helpers.GetTaskKey(pod)
+	masterIndex := helpers.GetTasklndexUnderJob(pp.masterName, job)
+	if masterIndex == -1 {
+		klog.Errorf("job %v doesn't have task %v", job.Name, pp.masterName)
```

Comment:
If the job contains no master pod, is it necessary still to open ports for all containers? Or just return the error?

Reply:
Yes you're right, I removed this logic